streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -410,8 +374,8 @@ public void close() {
             }
 
             // all threads should be stopped and cleaned up by now, so none should remain holding a lock
-            if (locks.isEmpty()) {
-                log.error("Some task directories still locked while closing state, this indicates unclean shutdown: {}", locks);
+            if (lockedTasksToStreamThreadOwner.isEmpty()) {
```

Comment:
Hmm.. should that actually be `!isEmpty())`? Were we actually always print this error message spuriously?

Reply:
Ahhh yeah it should. I kept wondering why I was seeing this sometimes in the soak or various unit tests, and just assumed it was because of an unclean shutdown 🤦‍♀️. Thanks for catching this